src/client/Microsoft.Identity.Client/ManagedIdentity/AppServiceManagedIdentitySource.cs

```diff
@@ -83,14 +80,12 @@ protected override ManagedIdentityRequest CreateRequest(string[] scopes)
                 _requestContext.Logger.Info("[Managed Identity] Adding user assigned client id to the request.");
                 request.QueryParameters[Constants.ManagedIdentityClientId] = _clientId;
             }
-
-            if (!string.IsNullOrEmpty(_resourceId))
+            else if (!string.IsNullOrEmpty(_resourceId))
```

Comment:
Why is the else needed? Are you not able to add both client id and resource id to the QueryParameters?